src/network/networkd-dhcp-prefix-delegation.c

```diff
@@ -869,8 +869,8 @@ static int dhcp4_pd_assign_subnet_prefix(Link *link, Link *uplink) {
         if (r < 0)
                 return log_link_warning_errno(uplink, r, "Failed to get lifetime of DHCPv4 lease: %m");
 
-        assert_se(sd_event_now(link->manager->event, CLOCK_BOOTTIME, &now_usec) >= 0);
-        lifetime_usec = usec_add(lifetime_sec * USEC_PER_SEC, now_usec);
+        assert_se(sd_event_now(uplink->manager->event, CLOCK_BOOTTIME, &now_usec) >= 0);
```

Comment:
this changes from link to uplink in the last commit, but doesn't seem related? same below